src/socket.c

```diff
@@ -3378,6 +3453,15 @@ static int handle_socket_input(const char *simbuffer, int simlen, const char *en
 #if HAVE_MCCP
 		    (rawchar == TN_COMPRESS && mccp) ||
 		    (rawchar == TN_COMPRESS2 && mccp) ||
+#endif
+#if ENABLE_ATCP
+		    (rawchar == TN_ATCP && atcp) ||
+#endif
+#if ENABLE_GMCP
+		    (rawchar == TN_GMCP && gmcp) ||
+#endif
+#if ENABLE_OPTION102
+		    (rawchar == TN_102 && 102) ||
```

Comment:
`&& 102` will never be false, but it looks like this should have been `&& option102`.